components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -1918,9 +1920,7 @@ private void setAuthorizationCode(OAuthMessage oAuthMessage, OAuth2AuthorizeResp
         authorizationResponseDTO.getSuccessResponseDTO().setAuthorizationCode(authorizationCode);
 
         AccessTokenExtendedAttributes tokenExtendedAttributes = null;
-        if (isConsentResponseFromUser(oAuthMessage)) {
-            tokenExtendedAttributes = getExtendedTokenAttributes(oAuthMessage, oauth2Params);
-        }
+        tokenExtendedAttributes = getExtendedTokenAttributes(oAuthMessage, oauth2Params);
```

Comment:
for each authz call service provider instance will be retrieve from FW, this is not ideal fix, But approve for now.

Reply:
We have to have the check whether the app has this function implemented in script. That's a must. We should not let this impact the 80% of the use cases